### ISSUE TITLE
Latest NRI-Bundle chart up to 5.0.17 breaks working installation

### ISSUE THREAD

**Issue by armenr** [newrelic/helm-charts#1106]
**Bug description**

Upgraded to the latest NRI-Bundle chart

version we were using: `5.0.15`
version we moved to: `5.0.17`

Before upgrade --> everything was working as expected.
After upgrade --> errors

## Errors from `newrelic-nrk8s-kubelet`
`newrelic-nrk8s-kubelet` starts to error out:

```
kubelet time="2023-06-05T08:01:10Z" level=info msg="Waiting for agent container to be ready..."
kubelet time="2023-06-05T08:01:15Z" level=info msg="New Relic Kubernetes integration Version: v3.13.0, Platform: linux/amd64, GoVersion: go1.19.9, GitCommit: fed53d76f4e295e5c11d92199f9bfde43ed3f3d7, BuildDate: Fri May 26 22:28:32 UTC 2023\n"
kubelet time="2023-06-05T08:01:15Z" level=info msg="Trying to connect to kubelet locally with scheme=\"https\" hostURL=\"172.17.18.232:10250\""
kubelet time="2023-06-05T08:01:15Z" level=info msg="Connected to Kubelet through nodeIP with scheme=\"https\" hostURL=\"172.17.18.232:10250\""
kubelet time="2023-06-05T08:01:46Z" level=warning msg="Error sending data to agent sink: {\"2023-06-05 08:01:46.033064113 +0000 UTC m=+35.213452831\" \"Do\" \"http://localhost:8003/v1/data\" \"POST\" '\\x00' '\\x02' '\\x01' \"Post \\\"http://localhost:8003/v1/data\\\": dial tcp [::1]:8003: connect: connection refused\"}"
agent time="2023-06-05T08:01:10Z" level=info msg="runtime configuration" agentUser=root component="New Relic Infrastructure Agent" executablePath= maxProcs=1 pluginDir="[/etc/newrelic-infra/integrations.d /var/db/newrelic-infra/integrations.d]"
agent time="2023-06-05T08:01:10Z" level=info msg="Checking network connectivity..." component=AgentService service=newrelic-infra
agent time="2023-06-05T08:01:11Z" level=info msg=Initializing component=AgentService elapsedTime=111.062889ms service=newrelic-infra version=1.42.1
agent time="2023-06-05T08:01:11Z" level=info msg="New Relic infrastructure agent is running." component=AgentService elapsedTime=637.041925ms service=newrelic-infra
agent time="2023-06-05T08:01:11Z" level=info msg="Starting up agent..." component=Agent
agent time="2023-06-05T08:01:11Z" level=info msg="Agent plugin" plugin=metadata/attributes
kubelet time="2023-06-05T08:01:47Z" level=warning msg="Error sending data to agent sink: {\"2023-06-05 08:01:47.034090672 +0000 UTC m=+36.214479390\" \"Do\" \"http://localhost:8003/v1/data\" \"POST\" '\\x00' '\\x03' '\\x02' \"Post \\\"http://localhost:8003/v1/data\\\": dial tcp [::1]:8003: connect: connection refused\"}"
kubelet time="2023-06-05T08:01:49Z" level=warning msg="Error sending data to agent sink: {\"2023-06-05 08:01:49.034607874 +0000 UTC m=+38.214996597\" \"Do\" \"http://localhost:8003/v1/data\" \"POST\" '\\x00' '\\x04' '\\x03' \"Post \\\"http://localhost:8003/v1/data\\\": dial tcp [::1]:8003: connect: connection refused\"}"
kubelet time="2023-06-05T08:01:49Z" level=error msg="publishing integration: performing HTTP request: Post \"http://localhost:8003/v1/data\": dial tcp [::1]:8003: connect: connection refused"
agent time="2023-06-05T08:01:11Z" level=info msg="Agent plugin" plugin=metadata/system
agent time="2023-06-05T08:01:11Z" level=info msg="Agent plugin" plugin=metadata/host_aliases
agent time="2023-06-05T08:01:11Z" level=info msg="Agent plugin" plugin=metadata/agent_config
agent time="2023-06-05T08:01:11Z" level=info msg="Agent plugin" plugin=metadata/proxy_config
agent time="2023-06-05T08:01:11Z" level=info msg="connect got id" agent-guid=MzM5NTE2fElORlJBfE5BfC0zMzU2NDkyMDc0MTQxNTg1NDkz agent-id=-3356492074141585493 component=IdentityConnectService
agent time="2023-06-05T08:01:31Z" level=warning msg="instantiating container sampler process decorator" component=ProcessSampler error="containerd sampler: no running task found: task 4e64d77139ba566b20410aacc69cc47effb4e657b771d205bc317ed7feed99dc not found: not found"
agent panic: runtime error: invalid memory address or nil pointer dereference
agent [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16aec2e]
agent
agent goroutine 143 [running]:
agent github.com/newrelic/infrastructure-agent/pkg/metrics.(*containerdDecorator).Decorate(0xc0005ce5e8?, 0xc000c49760)
agent     /go/src/github.com/newrelic/infrastructure-agent/pkg/metrics/containerd_sampler.go:173 +0x2e
agent github.com/newrelic/infrastructure-agent/pkg/metrics/process.(*processSampler).Sample(0xc000536f50)
agent     /go/src/github.com/newrelic/infrastructure-agent/pkg/metrics/process/sampler_linux.go:127 +0x3de
agent github.com/newrelic/infrastructure-agent/pkg/metrics/sampler.StartSamplerRoutine.func1.2({0x1d4a340, 0xc000536f50})
agent     /go/src/github.com/newrelic/infrastructure-agent/pkg/metrics/sampler/sampler_routine.go:49 +0x12c
agent github.com/newrelic/infrastructure-agent/pkg/metrics/sampler.StartSamplerRoutine.func1()
agent     /go/src/github.com/newrelic/infrastructure-agent/pkg/metrics/sampler/sampler_routine.go:50 +0x18a
agent created by github.com/newrelic/infrastructure-agent/pkg/metrics/sampler.StartSamplerRoutine
agent     /go/src/github.com/newrelic/infrastructure-agent/pkg/metrics/sampler/sampler_routine.go:35 +0x16e
```

**Version of Helm and Kubernetes**

K8s version: `K8s Rev: v1.27.1-eks-2f008fe`
Helm version: `version.BuildInfo{Version:"v3.12.0", GitCommit:"c9f554d75773799f72ceef38c51210f1842a1dea", GitTreeState:"clean", GoVersion:"go1.20.4"}`

**Which chart?**

`name: nri-bundle`
`version: 5.0.17`

**What happened?**

Upgrade `nri-bundle` chart from `5.0.15` --> `5.0.17` (nothing else changed at all)

**What you expected to happen?**

Expected for newrelic `nri-bundle` umbrella chart to continue working as it had before (we had a working installation)

**How to reproduce it?**

### `values.yaml`

```
global:
  licenseKey: REDACTED
  cluster: REDACTED
newrelic-prometheus-agent:
  enabled: true
  config:
    kubernetes:
      integrations_filter:
        enabled: false
newrelic-pixie:
  enabled: false
pixie-chart:
  enabled: false
nri-kube-events:
  enabled: true
newrelic-logging:
  enabled: true
kube-state-metrics:
  enabled: true
  image:
    tag: v2.8.2
newrelic-infrastructure:
  enabled: true
newrelic-infrastructure:
  privileged: true
kubeEvents:
  enabled: true

```

**Anything else we need to know?**

I've set the Security Group for the eks worker nodes to allow all traffic from self --> to self (no port or traffic restriction between any of the worker nodes or the pods in those worker nodes).

Traffic between worker nodes and "primary" nodes (control plane) is explicitly controlled.

We run our cluster using Cilium, with a `tunnel` datapath (`vxlan` overlay), instead of using the `aws-vpc-cni`, and we *do not* use Cilium's `eni` datapath either. 

We are not currently using any Kubernetes or Cilium based Network Policies to restrict or control traffic inside the cluster, either.

**Comment by jodem** [comment]
Same problem here with kube 1.26 on EKS after upgrading from an 4.x version.

**Comment by nipr-jdoenges** [comment]
This seems to be the same issue reported for newrelic-infrastructure chart version 3.18.0 https://github.com/newrelic/nri-kubernetes/issues/774

**Comment by nipr-jdoenges** [comment]
Experienced this with chart version 5.0.16 as well.

**Comment by najlamazouz** [comment]
I think it is fixed in 5.0.18
